Formula/scorecard.rb

```diff
@@ -0,0 +1,23 @@
+class Scorecard < Formula
+  desc "Security health metrics for Open Source"
+  homepage "https://github.com/ossf/scorecard"
+  url "https://github.com/ossf/scorecard/archive/refs/tags/v2.0.0.tar.gz"
+  sha256 "9d3f0dd0a82cdc4cb3cd69149ad0967291b1bafb751bb07bc9c92db00af613a9"
+  license "Apache-2.0"
+  head "https://github.com/ossf/scorecard.git"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["GO111MODULE"] = "on"
+
+    system "make", "install"
+    system "make", "build-scorecard"
+    bin.install "scorecard"
+  end
+
+  test do
+    system "#{bin}/scorecard", "help"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Apologies, trying to get a good test working now. I missed that in the guide as I've seen quite a lot of tests which just have `--help` or `--version` calls.

Trying to do a test call of the executable, however it requires a `GITHUB_AUTH_TOKEN` which might cause issues. So I've set it to an invalid value to check that it fails as expected.